### DISCUSSION TITLE
Make `ae` available:    pharmaversesdtm::ae   vs library(pharmaversesdtm)   - a dead horse?

### DISCUSSION THREAD

**Discussion by jimrothstein** [pharmaverse/admiral#2518] [General]
Know there has been much discussion on this.   But still not clear to me why admiral uses  BOTH:
### library(pharmaversesdtm) and pharmversesdtm::ae

`library` seems enough to me:

(I can clean this up, but here is my scratch work:)
  GOAL:   make dataset `ae` available for use
  COMPARE:  library(pharmaversesdtm)   vs pharmaversesdtm::ae

`if pkg installed,` this will find `ae` even if package is not loaded
```
reset()  #  explained below
data(package = lib_sdtm)    # locates `ae`, but you can use it yet!
```

**METHOD 1 to make ae available.**

Call library on pkg `pharmaversesdtm`
I can not spell `pharmaversesdtm`; so trick to evaluate the string
```
lib_sdtm = "pharmaversesdtm"
library(lib_sdtm, character.only=T)
ae    # **now found, our goal**
```

## Tool to clean out packages, NS and so ae is not found
```
reset= function() {
  unloadNamespace("pharmaversesdtm")
}


reset()
```
### check
```
isNamespaceLoaded(lib_sdtm)   # all negative
search()  # no
ae  # not found
getAnywhere(ae)   # not found
```

### Method 2 to make ae available, use pharmaversesdtm::ae EACH time
###(but does not attach the package)
```
pharmaversesdtm::ae   # loads NS and ae is available

### check - `ae` not otherwise present,  `must` use `::` each time
isNamespaceLoaded(lib_sdtm)   # loaded not attached
search()                      # not attached
ae                            # not found
getAnywhere(ae)
```

So both methods make `ae` available,  but do we need both.

**Comment by bms63** [comment]
I guess we are just trying to be explicit on where the data is coming from as we are assuming folks are not super familiar with R yet if they are using our template programs.

Yes we could drop the library() call, but the comment on the datasets origin is important to also point out to new users.

**Reply by jimrothstein** [reply]
Yes, you did say purpose of `pharamaversesdtm::ae` is to be explicit.
- I will add this to the FAQ I am (slowly) keeping.
- IMHO, new programmer would benefit from enhancing the comments.   But if that is not  your view/consensus, I drop it since there are other things to do.   Just asking.
-  In particular,  as an example
```
# Package pharmaversesdtm provides necessary datasets (ae, dm ...) for this vignette, template, ...
library(pharmaversesdtm)
```
